asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
+Together with the constraints that define the ARM memory model~\cite{AlglaveDGHM21}, these graphs axiomatically define
+the valid sequences of read and write operations induced by a given specification.
+
+\paragraph{Sequential Semantics.}
+The sequential semantics corresponds to executing an ASL specification in the context of a single (conceptual) process
+of execution --- notice that ASL does not contain any concurrency constructs. Access to memory is hidden away
+in primitive subprograms that interface with a map-like memory implementation that does not reorder memory accesses
+and serves one access at a time.
+%
+Technically, the sequential semantics is defined by dropping the \herd\ execution graph components
+from the configurations.
+
+\section{Mathematical Notations and Conventions}
+
+\input{ASLNotations.tex}
+
+\paragraph{OCaml-style Notations.}
+We use the notation $L(v_{1..k})$, where $L$ is a label and $v_{1..k}$ is a (possibly-singleton) tuple of abstract values,
+to denote the tuple $(L,v_{1..k})$.
+We also write $L(T_{1..k})$, where $T_{1..k}$ denote mathematical type of values, to stand for
+$\{L\} \times T_1 \times \ldots \times T_k$.
+%
+We later use these notations to define \emph{semantic configurations} and \emph{native values}, \underline{not to be confused with labelled AST nodes}.
+%
+The notation $\langle \cdot \rangle$ stands for empty-or-singleton sets, where $\None\triangleq\langle\rangle$ denotes
+an empty set and $\langle v \rangle$ denotes a set containing the single element $v$.
+%
+The notation $\langle T \rangle$, where $T$ denotes a mathematical type of values, stands for
+$\{ \langle\rangle \} \cup \{\langle v \rangle \;|\; v \in T\}$.
+
+\paragraph{Misclleaneous.}
+We define the following shorthand notations for the true and false expressions:
+\newcommand\etrue[0]{\texttt{ETrue}}
+\newcommand\efalse[0]{\texttt{EFalse}}
+\[
+\begin{array}{rcl}
+  \etrue &\triangleq& \texttt{E\_Literal}(\texttt{L\_Bool}(\texttt{True}))\\
+  \efalse &\triangleq& \texttt{E\_Literal}(\texttt{L\_Bool}(\texttt{False}))\\
+\end{array}
+\]
+
+\section{Configurations}
+
+Configurations express intermediate states related by \emph{semantic relations}.
+%
+More precisely, semantic relations relate two distinct sets of configurations ---
+\emph{input configurations} and \emph{output configurations}.
+Input configurations consist of an environment and an AST node.
+Output configurations consist of an output environment, values,
+and \herd\ execution graphs.
+%
+Configurations wrap together elements such as environments and AST nodes
+and associate them with a \emph{mode}. Input configurations use modes
+that determine the semantic relation they pertain to, while output configurations
+use modes to distinguish between conceptually different kinds of outputs, for example
+ones where an exception was raised, ones when a dynamic error occurred, etc.
+
+We now explain the components over which configurations are defines.
```

Comment:
typo: defines -> defined
also please give here a bullet point list of what those components are, with a \label \ref reference to the corresponding section, and a short description of what they are. Something like this:
- Dynamic Environments (see Section~\ref{sec:DynamicEnvironments}), which associates native values to variables;
- etc


Also please wrap the section about Native Values within the Dynamic Environments one - since Native Values re not a configuration component. Essentially what I would like is that the subsection level at this point only is about the components configurations.

Reply:
First comment - done.
Regarding native values, they also appear separately, outside of any environment.

Reply:
I understand that - I am asking that the structure of this specific part of the document reflects the components of semantics configurations. Please could you define Native Values elsewhere, in a section of their own?